Allow unsigned boolean declarations

succinct opened the pull request:
Update getBooleanTypeDeclarationSQL() to allow UNSIGNED

succinct committed:
Allow unsigned boolean declarations

Update getBooleanTypeDeclarationSQL() to allow UNSIGNED

deeky666 commented on lib/Doctrine/DBAL/Platforms/MySqlPlatform.php:
```diff
@@ -289,7 +289,7 @@ public function getTimeTypeDeclarationSQL(array $fieldDeclaration)
     public function getBooleanTypeDeclarationSQL(array $field)
     {
-        return 'TINYINT(1)';
+        return 'TINYINT(1)' . $this->_getCommonIntegerTypeDeclarationSQL($field);;
```
This is akward on MySQL. In the past we used to decline this kind of request. But I get the desire. Please just use `'UNSIGNED'` here. No need to evaluate any other properties of integer type columns here. Also declaring `unsigned` property on boolean type columns should not be allowed. Also please add a testcase for this. Thanks.

succinct committed:
Updated UNSIGNED BOOLEAN implementation

Restructured code to allow UNSIGNED declarations without also including AUTO_INCREMENT.

succinct commented:
I decided it's better to fix our schema to make all boolean signed, rather than bother with this change.
